### PR TITLE
debian: cleanups and exclude ceph_crypto from dh_shlibdep

### PR DESCRIPTION
**Pull request by tchaikov**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by tchaikov** [comment]
jenkins test make check

**Comment by tchaikov** [comment]
- https://pulpito.ceph.com/kchai-2021-03-20_13:57:49-rados-wip-kefu2-testing-2021-03-20-1320-distro-basic-smithi/
- https://pulpito.ceph.com/kchai-2021-03-22_02:52:16-rados-wip-kefu2-testing-2021-03-21-2353-distro-basic-smithi/

failures tracked by

- https://tracker.ceph.com/issues/49916
- https://tracker.ceph.com/issues/49909

File: debian/control [hunk 1]
```diff
@@ -50,7 +49,7 @@ Build-Depends: automake,
                libldap2-dev,
                libleveldb-dev,
                liblttng-ust-dev,
-               liblua5.3-dev,
+               liblua5.3-dev | liblua5.4-dev,
```
Comment (yuvalif) [inline debian/control:52]: why is this needed, both ubuntu focal and bionic use liblua5.3 ?

Comment (tchaikov) [reply debian/control:52]: yeah, i am scratching my own itch. as on debian sid, quite a few packages are already compiled against liblua5.4. for instance, see https://packages.debian.org/sid/neomutt . so i didn't want to install liblua5.3-dev for compiling ceph since i've installed liblua5.4-dev. if you are still concerned about this change, i will be happy to drop it.

Comment (yuvalif) [reply debian/control:52]: from thew lua RGW binding side of things i don't think it makes a difference - lua5.4 is backward compatible, and [the changes](https://www.lua.org/manual/5.4/readme.html#changes) should not impact it anyway.
but I'm not 100% regarding luarocks (which is also our dependency) which has a lua5.3 dependency even in its [latest version](https://packages.debian.org/sid/luarocks).

this could be easily tested, even without the RGW - just make sure that luarocks work on a debian sid machine with lua5.4 installed on.

Comment (tchaikov) [reply debian/control:52]: > this could be easily tested, even without the RGW - just make sure that luarocks work on a debian sid machine with lua5.4 installed on.

```
pi   --\ luarocks                                                                                                                                                                                                                        +705 kB   <none>         2.4.2+dfsg-1.1
  --\ Depends (6)
    --- liblua5.1-dev | liblua5.2-dev | liblua5.3-dev
    --- lua-any
    --\ lua5.1 | lua5.2 | lua5.3
p     lua5.1 5.1.5-8.1+b3
i     lua5.2 5.2.4-1.1+b3
p     lua5.3 5.3.3-1.1+b1
```

ahh, then i have to drop this change.

**Comment by tchaikov** [comment]
@yuvalif dropped and repushed.

**Review by yuvalif — APPROVED** [review]

**Comment by djgalloway** [comment]
@tchaikov how does the `with_system_libs` var work?  I think it needs to be true if we're building on Ubuntu and the version is Jammy or later.

We're seeing https://tracker.ceph.com/issues/54473 but when building for Ubuntu 22.04.

I feel like it should get set as a default in `debian/rules` (instead of in ceph-build.git scripts/build_utils.sh) but I'm in over my head.

**Comment by tchaikov** [comment]
@djgalloway unless we build the static libraries for the homebrew dependency packages which are not provided by the official distro (Ubuntu focal in this case), we cannot enable this option. Otherwise, the built packages would have a runtime dependency on these homebrew packages, hence user would not be able to even install them without adding our corresponding chacra repo beforehand.

**Comment by djgalloway** [comment]
I think the error means we /want/ to use the distro-provided pmdk, no?

**Comment by tchaikov** [comment]
@djgalloway sorry, i missed the context of your question. yes, i think so.

> @tchaikov how does the `with_system_libs` var work? I think it needs to be true if we're building on Ubuntu and the version is Jammy or later.

just
```bash
export DEB_BUILD_OPTIONS=with_system_libs
```

see https://www.debian.org/doc/debian-policy/ch-source.html#s-debianrules-options

**Comment by djgalloway** [comment]
But that's something that should be set in `debian/rules` itself, right?  Not in ceph-build.git?  We want folks to be able to build Ceph on Ubuntu 22.04 without knowing they need to add `DEB_BUILD_OPTIONS=with_system_libs` I would presume?

**Comment by djgalloway** [comment]
Like, I know I can easily add an `if ubuntu jammy or later; then export DEB_BUILD_OPTIONS=with_system_libs; fi` statement in ceph-build scripts/build_utils.sh but then `make` would still fail with just the ceph.git checkout on Ubuntu 22.04.

**Comment by tchaikov** [comment]
quoting my messages in `#sepia` channel, in case they were missed:

> 
> [21:54:24]  <kefu> just set the env var, and pbuilder or dpkg-build will pick it up.
> [21:54:39]  <kefu> no need to touch the debian/rules.
> [21:55:03]  <kefu> s/dpkg-build/dpkg-buildpackage/
> [21:55:21]  <kefu> >  i just don't know how to do that in ceph.git based on the OS version
> [21:55:32]  <kefu> oh, will look at your q more closely.
> [21:59:47]  <kefu> dgalloway, as you probably already know, we build debian packages using pbuilder.
> [22:00:17]  <kefu> and we use `setup_pbuilder()` for setting up pbuilder options.
> [22:03:57]  <kefu> and in build_debs() we populate the env variables to pbuilder
> [22:04:25]  <kefu> see https://github.com/ceph/ceph-build/blob/0049e3f3a4af6d23bc8bb173f3e98b5e2b4dd6e5/scripts/build_utils.sh#L915
> [22:04:39]  <kefu> and https://github.com/ceph/ceph-build/blob/0049e3f3a4af6d23bc8bb173f3e98b5e2b4dd6e5/scripts/build_utils.sh#L918-L927

**Comment by djgalloway** [comment]
Octopus `make check` fails on Jammy :roll_eyes:

https://jenkins.ceph.com/job/ceph-pull-requests/100740/console

**Comment by tchaikov** [comment]
@djgalloway #47448

**Comment by djgalloway** [comment]
Makes sense.  Thank you.
